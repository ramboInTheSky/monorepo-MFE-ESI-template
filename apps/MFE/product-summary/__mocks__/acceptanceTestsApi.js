// Test Suit for product with 20 colourways

exports.testProductItemNumber = "TEST_20_COLOURWAYS"
exports.testProductTitle = "Denim Crew Neck T-shirt"
exports.testColourway1ItemNumber = "1111111"
exports.testColourway1Colour = "Dark"
exports.testColourway1Url = "g7190s5/962950#111111"
exports.testPrice = "£10 - £12.50"

exports.testColourwayViewMoreItemNumber = "19"

exports.apiData = {
    itemNumber: exports.testProductItemNumber,
    title: exports.testProductTitle,
    currencyCode: "GBP",
    colourways: [
        {
            fits: ["Regular", "Petite"],
            url: exports.testColourway1Url,
            colour: exports.testColourway1Colour,
            itemNumber: exports.testColourway1ItemNumber,
            title: exports.testColourway1Colour,
            overallStarRating: 2,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12.5,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Bleach Wash",
            itemNumber: "920427",
            title: exports.testColourway1Colour,
            overallStarRating: 4,
            itemPrice: {
                minPrice: 10,
                maxPrice: 10,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "503290",
            title: exports.testColourway1Colour,
            overallStarRating: 4,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: 1,
                saleMaxPrice: 2,
                wasMinPrice: 10,
                wasMaxPrice: 12,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "940736",
            title: exports.testColourway1Colour,
            overallStarRating: 5,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: 5,
                saleMaxPrice: 5,
                wasMinPrice: 10,
                wasMaxPrice: 10,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "495341",
            title: exports.testColourway1Colour,
            overallStarRating: 3,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "507422",
            title: exports.testColourway1Colour,
            overallStarRating: 1,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "608072",
            title: exports.testColourway1Colour,
            overallStarRating: 0,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "705031",
            title: exports.testColourway1Colour,
            overallStarRating: 3,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "803745",
            title: exports.testColourway1Colour,
            overallStarRating: 4,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "630710",
            title: exports.testColourway1Colour,
            overallStarRating: 2,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "326828",
            title: exports.testColourway1Colour,
            overallStarRating: 1,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "768334",
            title: exports.testColourway1Colour,
            overallStarRating: 5,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "966813",
            title: exports.testColourway1Colour,
            overallStarRating: 3,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "745348",
            title: exports.testColourway1Colour,
            overallStarRating: 2,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "969758",
            title: exports.testColourway1Colour,
            overallStarRating: 2,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "819434",
            title: exports.testColourway1Colour,
            overallStarRating: 2,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "466748",
            title: exports.testColourway1Colour,
            overallStarRating: 2,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "313332",
            title: exports.testColourway1Colour,
            overallStarRating: 2,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: exports.testColourwayViewMoreItemNumber,
            title: exports.testColourway1Colour,
            overallStarRating: 2,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "757786",
            title: exports.testColourway1Colour,
            overallStarRating: 2,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "999204",
            title: exports.testColourway1Colour,
            overallStarRating: 2,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
    ],
}
