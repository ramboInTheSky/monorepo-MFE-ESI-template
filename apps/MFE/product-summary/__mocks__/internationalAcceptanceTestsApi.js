// Test Suit for product with 20 colourways

exports.testProductItemNumber = "INTERNATIONAL_TEST_20_COLOURWAYS"
exports.testProductTitle = "Matt Emulsion 100ml Tester Pot"
exports.testColourway1ItemNumber = "1111111"
exports.testColourway1Title = "Matt Emulsion"
exports.testColourway1Colour = "Dark"
exports.testColourway1Url = "g7190s5/962950#111111"
exports.testPrice = "10 د.إ‏ - 12.50 د.إ‏"

exports.testColourwayViewMoreItemNumber = "19"

exports.apiData = {
    itemNumber: exports.testProductItemNumber,
    title: exports.testProductTitle,
    currencyCode: "AED",
    brand: "nike",
    department: "menswear",
    fit: "tall",
    colourways: [
        {
            fits: ["Regular", "Petite"],
            url: exports.testColourway1Url,
            colour: exports.testColourway1Colour,
            itemNumber: exports.testColourway1ItemNumber,
            title: exports.testColourway1Title,
            overallStarRating: 3,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12.5,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Bleach Wash",
            itemNumber: "920427",
            title: exports.testColourway1Title,
            overallStarRating: 2,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: 3,
                saleMaxPrice: 4,
                wasMinPrice: 5,
                wasMaxPrice: 6,
            },
        },
        {
            fits: ["Regular"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "503290",
            title: exports.testColourway1Title,
            overallStarRating: 4,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "940736",
            title: exports.testColourway1Title,
            overallStarRating: 5,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "495341",
            title: exports.testColourway1Title,
            overallStarRating: 1,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "507422",
            title: exports.testColourway1Title,
            overallStarRating: 0,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "608072",
            title: exports.testColourway1Title,
            overallStarRating: 4,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "705031",
            title: exports.testColourway1Title,
            overallStarRating: 3,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "803745",
            title: exports.testColourway1Title,
            overallStarRating: 5,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "630710",
            title: exports.testColourway1Title,
            overallStarRating: 3,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "326828",
            title: exports.testColourway1Title,
            overallStarRating: 2,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "768334",
            title: exports.testColourway1Title,
            overallStarRating: 1,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "966813",
            title: exports.testColourway1Title,
            overallStarRating: 0,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "745348",
            title: exports.testColourway1Title,
            overallStarRating: 3,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "969758",
            title: exports.testColourway1Title,
            overallStarRating: 3,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "819434",
            title: exports.testColourway1Title,
            overallStarRating: 3,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "466748",
            title: exports.testColourway1Title,
            overallStarRating: 3,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "313332",
            title: exports.testColourway1Title,
            overallStarRating: 3,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: exports.testColourwayViewMoreItemNumber,
            title: exports.testColourway1Title,
            overallStarRating: 3,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "757786",
            title: exports.testColourway1Title,
            overallStarRating: 3,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
        {
            fits: ["Regular/Tall", "Petite"],
            url: "g7190s5/962950#962950",
            colour: "Dark",
            itemNumber: "999204",
            title: exports.testColourway1Title,
            overallStarRating: 3,
            itemPrice: {
                minPrice: 10,
                maxPrice: 12,
                saleMinPrice: null,
                saleMaxPrice: null,
                wasMinPrice: null,
                wasMaxPrice: null,
            },
        },
    ],
}
