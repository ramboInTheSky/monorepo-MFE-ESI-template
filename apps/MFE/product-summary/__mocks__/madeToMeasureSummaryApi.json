{
  "itemNumber": "113364",
  "title": "Joules Black Loxley Waterproof Padded Technical Raincoat",
  "brand": null,
  "department": null,
  "fit": null,
  "currencyCode": "GBP",
  "colourways": [
    {
      "fits": [],
      "itemNumber": "113364",
      "title": "Joules Black Loxley Waterproof Padded Technical Raincoat",
      "colour": null,
      "url": "",
      "itemPrice": {
        "minPrice": 55,
        "maxPrice": 90,
        "saleMinPrice": null,
        "saleMaxPrice": null,
        "wasMinPrice": null,
        "wasMaxPrice": null
      },
      "overallStarRating": 4,
      "personalisedType": "madeToMeasure"
    }
  ]
}
