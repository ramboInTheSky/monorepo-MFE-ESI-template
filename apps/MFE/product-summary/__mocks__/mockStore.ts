import configureMockStore from "redux-mock-store"
import {mockColors} from "@monorepo/themes"
import State from "../src/models/state"
import {Fits, ProductType, SofaType} from "../src/config/constants"
import text from "./default-text.json"

export const mockState: State = {
    text,
    textAlignment: "ltr",
    productSummary: {
        summaryData: {
            title: "My Product",
            id: "00000",
            brand: "nike",
            department: "menswear",
            fit: "tall",
            currencyCode: "GBP",
            baseUrl: "http://test.co.uk",
            imageCdnUrl: "http://testcdn.co.uk",
            productImageUrlPart: "search",
            showNewIn: false,
            saleSash: null,
            saleSashPosition: null,
            colourways: [
                {
                    title: "White",
                    fits: [Fits.Tall],
                    id: "99434296",
                    colour: "White",
                    url: "g5990s21/845632#845632",
                    selected: true,
                    price: "£123",
                    salePrice: null,
                    wasPrice: null,
                    minPrice: "£123",
                    overallStarRating: 0,
                    colourChipImage: "http://testcdn.co.uk/AltItemSwatch/21x21/99434296.jpg",
                    mainImage: "http://testcdn.co.uk/search/224x336/99434296.jpg",
                    personalisedType: "Test",
                },
                {
                    title: "Black",
                    fits: [Fits.Tall, Fits.Petite],
                    id: "434296",
                    colour: "Black",
                    url: "g5990s21/845632#845632",
                    selected: false,
                    price: "£99",
                    salePrice: null,
                    wasPrice: null,
                    minPrice: "£99",
                    overallStarRating: 4,
                    colourChipImage: "http://testcdn.co.uk/AltItemSwatch/21x21/434296.jpg",
                    mainImage: "http://testcdn.co.uk/search/224x336/434296.jpg",
                    personalisedType: "Test",
                },
            ],
            type: ProductType,
        },
        canLoadVisibleColourways: false,
        hasLoadedVisibleColourways: false,
        isFav: false,
        enableFav: false,
        enableReviewStars: false,
        showFavErrorToolTip: null,
        enabledSearchDesc: false,
        isLoadingFav: false,
        animateFavouriteIcon: false,
        enableBrandDisplay: false,
    },
    lazyload: {
        colourchips: false,
        productImages: false,
        fitIcons: false,
    },
    request: {
        headers: {"x-monorepo-language": "en"},
        isEnglishLang: true,
    },
}

export const mockStateSofa: State = {
    ...mockState,
    productSummary: {
        ...mockState.productSummary,
        summaryData: {
            title: "Jackson",
            id: "jackson_82_262919",
            brand: "",
            department: "homeware",
            fit: "",
            currencyCode: "GBP",
            baseUrl: "http://test.co.uk",
            imageCdnUrl: "http://testcdn.co.uk",
            productImageUrlPart: "search",
            showNewIn: false,
            saleSash: null,
            saleSashPosition: null,
            type: SofaType,
            colourways: [
                {
                    title: "Sofa 1",
                    fits: [],
                    id: "jackson_82_262919",
                    colour: "Chunky Weave/Dove",
                    url: "/homeware/sofas-chairs/jackson/size-power-recliner-large-sofa-fabric-chunky-weave-dove",
                    selected: true,
                    price: "£775 - £1,350",
                    salePrice: null,
                    wasPrice: null,
                    minPrice: "£775",
                    mainImage: "http://testcdn.co.uk/262919_82.jpg",
                    colourChipImage: "http://testcdn.co.uk/SofaSwatchView/chunkyweave_dove.jpg",
                    overallStarRating: 0,
                    personalisedType: "",
                },
                {
                    title: "Sofa 2",
                    fits: [],
                    id: "jackson_82_202546",
                    colour: "Monza Faux Leather/Charcoal",
                    url: "/homeware/sofas-chairs/jackson/size-power-recliner-large-sofa-fabric-monza-faux-leather-charcoal",
                    selected: false,
                    price: "£775 - £1,350",
                    salePrice: null,
                    wasPrice: null,
                    minPrice: "£775",
                    mainImage: "http://testcdn.co.uk/202546_82.jpg",
                    colourChipImage: "http://testcdn.co.uk/SofaSwatchView/monzafauxleather_charcoal.jpg",
                    overallStarRating: 0,
                    personalisedType: "",
                },
                {
                    title: "Sofa 3",
                    fits: [],
                    id: "jackson_82_286382",
                    colour: "Matt Velvet/Navy Jackson",
                    url: "/homeware/sofas-chairs/jackson/size-power-recliner-large-sofa-fabric-matt-velvet-navy",
                    selected: false,
                    price: "£775 - £1,350",
                    salePrice: null,
                    wasPrice: null,
                    minPrice: "£775",
                    mainImage: "http://testcdn.co.uk/286382_82.jpg",
                    colourChipImage: "http://testcdn.co.uk/SofaSwatchView/mattvelvet_navy.jpg",
                    overallStarRating: 0,
                    personalisedType: "",
                },
            ],
        },
    },
}

export const mockTextAlignment = {Value: "ltr"}

export const mockTheme = {colours: mockColors, direction: "ltr", enablingCenteredContent: false}
export const centralisedMockTheme = {colours: mockColors, direction: "ltr", enablingCenteredContent: true}
const mockConfigureStore = configureMockStore()
const mockStore = mockConfigureStore(mockState)

export default mockStore
