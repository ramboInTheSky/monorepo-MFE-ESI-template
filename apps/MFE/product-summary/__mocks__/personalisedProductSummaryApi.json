{
  "itemNumber": "128638",
  "title": "Personalised  Navy Blue Crew & Joggers Jersey Set",
  "type": "Product",
  "currencyCode": "GBP",
  "brand": "Next",
  "department": "Childrenswear",
  "fit": null,
  "colourways": [
    {
      "fits": [],
      "title": "Personalised  Navy Blue Crew & Joggers Jersey Set",
      "personalisedType": "personalisedGift",
      "itemNumber": "128638",
      "colour": null,
      "url": "style/ST710416/128638#128638",
      "itemPrice": {
        "minPrice": 15,
        "maxPrice": 17,
        "saleMinPrice": null,
        "saleMaxPrice": null,
        "wasMinPrice": null,
        "wasMaxPrice": null
      },
      "overallStarRating": 3
    }
  ],
  "_rid": "l5NGAJWLW3NHAAAAAAAAAA==",
  "_self": "dbs/l5NGAA==/colls/l5NGAJWLW3M=/docs/l5NGAJWLW3NHAAAAAAAAAA==/",
  "_etag": "\"0c00a9e1-0000-0c00-0000-5ebe46160000\"",
  "_attachments": "attachments/",
  "_ts": 1589528086
}
