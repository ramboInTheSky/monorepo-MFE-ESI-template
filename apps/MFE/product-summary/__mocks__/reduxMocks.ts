import * as ReduxModule from "react-redux"

export function mockUseDispatchReturnValue(func = jest.fn()) {
    return jest.spyOn(ReduxModule, "useDispatch").mockReturnValue(func)
}
