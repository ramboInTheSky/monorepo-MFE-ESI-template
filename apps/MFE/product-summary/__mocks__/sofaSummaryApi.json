{
  "itemNumber": "jackson_82_262919",
  "title": "Jackson",
  "type": "sofa",
  "currencyCode": "GBP",
  "colourways": [
    {
      "itemNumber": "jackson_82_262919",
      "colour": "Chunky Weave/Dove",
      "url": "/homeware/sofas-chairs/jackson/size-power-recliner-large-sofa-fabric-chunky-weave-dove",
      "mainImage": "262919_82.jpg",
      "colourChipImage": "chunkyweave_dove.jpg",
      "itemPrice": {
        "minPrice": 775,
        "maxPrice": 1350,
        "saleMinPrice": 600,
        "saleMaxPrice": 600,
        "wasMinPrice": 1000,
        "wasMaxPrice": 1000
      }
    },
    {
      "itemNumber": "jackson_82_202546",
      "colour": "Monza Faux Leather/Charcoal",
      "url": "/homeware/sofas-chairs/jackson/size-power-recliner-large-sofa-fabric-monza-faux-leather-charcoal",
      "mainImage": "202546_82.jpg",
      "colourChipImage": "monzafauxleather_charcoal.jpg",
      "itemPrice": {
        "minPrice": 775,
        "maxPrice": 1350,
        "saleMinPrice": 600,
        "saleMaxPrice": 700,
        "wasMinPrice": 1000,
        "wasMaxPrice": 1200
      }
    },
    {
      "itemNumber": "jackson_82_286382",
      "colour": "Matt Velvet/Navy Jackson",
      "url": "/homeware/sofas-chairs/jackson/size-power-recliner-large-sofa-fabric-matt-velvet-navy",
      "mainImage": "286382_82.jpg",
      "colourChipImage": "mattvelvet_navy.jpg",
      "itemPrice": {
        "minPrice": 775,
        "maxPrice": 1350,
        "saleMinPrice": null,
        "saleMaxPrice": null,
        "wasMinPrice": null,
        "wasMaxPrice": null
      }
    }
  ]
}
