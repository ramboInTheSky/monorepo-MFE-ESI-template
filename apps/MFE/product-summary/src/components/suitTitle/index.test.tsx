import React from "react"
import {render} from "@testing-library/react"
import {ThemeProvider} from "styled-components"
import {mockTheme} from "../../../__mocks__/mockStore"
import {SuitTitle} from "."

describe("Price: ", () => {
    const props = {
        title: "Black batman suit",
        linkUrl: "www.test.com",
        tooltipTitle: "test title",
    }

    it("should match the snapshot ", () => {
        const {asFragment} = render(
            <ThemeProvider theme={mockTheme}>
                <SuitTitle {...props} />
            </ThemeProvider>,
        )
        expect(asFragment()).toMatchSnapshot()
    })
})
