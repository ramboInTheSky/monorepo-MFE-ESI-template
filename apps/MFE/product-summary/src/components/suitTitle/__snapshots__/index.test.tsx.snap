// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price:  should match the snapshot  1`] = `
<DocumentFragment>
  .c0 {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  display: block;
  text-transform: capitalize;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  font-family: "AzoSansRegular","Helvetica Neue",Arial,sans-serif;
  -webkit-letter-spacing: 0.008rem;
  -moz-letter-spacing: 0.008rem;
  -ms-letter-spacing: 0.008rem;
  letter-spacing: 0.008rem;
  line-height: 1.313rem;
  margin-bottom: 0.25rem;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover c0 MuiTypography-body1 MuiTypography-colorPrimary"
    data-desc="Black batman suit"
    data-label="Black batman suit"
    data-testid="suit_summary_title"
    data-usespan="False"
    href="www.test.com"
    title="test title"
  >
    Black batman suit
  </a>
</DocumentFragment>
`;
